### DISCUSSION TITLE
5.7.43 seems to have broke node

### DISCUSSION THREAD

**Discussion by KC5VDJ** [kej715/DtCyber#71] [General]
When trying to load the bunzip2 library.  For node start, install, etc..  even upgraded node and chocolaty last night to see if that was it, nope.

Also, interestingly, the existing 865 install ran fine as an 875 until it came to cybis...Eq05 seems to be a secured device, and will hang cybis from completing it's load, hang checkpoint system, and hang mag after checkpoint as well, giving a b,a side operator message about the secured device..  Tied to machine type?  I could have sworn install full compiled cybis, but after three hours of install, it does kinda blur a bit...gives a new appreciation of the ops, analysts, and engineers who had to babysit an install on the original (how long did it take at original speed?)

Reverting to .41 as an 865 shows nothing wrong with cybis using the exact images that bomb under .43 as an 875.  I could try 865 under .43, but I'm guessing it will work.  Academic anyhow until the node code is fixed, and an install can be achieved.

P.s. under nos 2.8.7

**Comment by KC5VDJ** [comment]
DtCyber run by hand, node non-functional, model=CYBER875
CM=20000000B
Run under unmodified 865 CM=4000000B install full.
Computer name wasn't changed to reflect 875 due to node being non-functional.

![sR0D1Xjlf3.png](https://github.com/kej715/DtCyber/assets/136194499/268ac65c-b67f-45ca-bfd8-b33220b646f2)

**Comment by kej715** [comment]
Please provide more information. I have not been able to reproduce the
problem. After cloning a fresh copy of the repo and running "make
Makefile,xxx all", then executing "sudo node install.js", the installation
completed successfully and CYBIS started and ran successfully too. After
changing the model to CYBER875, the system deadstarted successfully, and
CYBIS started and ran successfully as well. On what platform are you trying
to install DtCyber? Did you define a site.cfg file? If so, please provide a
copy, You implied that node has broken. By that, do you mean jode.js? If
so, what version of node.js are you running?

**Reply by KC5VDJ** [reply]
Ok.  I literally grabbed a zip of main.  Only copied over cyber.ini (after diff only showed my changes of machine name), and site.cfg only changed MID, NAME, and hostID.  Node.js MSI x64 win 10 (most recent updates, winblowz today, node upgraded last night), Thinkpad with specs in screenshot.

After an in-place unzip, only after adding the tapes to the zip, did dtcyber.sln.  I used defaults, mickeysoft backend, cleaned then built solution, clicked my shortcut, flash, nothing.

Ran by hand node start, traceback about loading unbzip2.

Did various combos, same.

Deleted entire dir, unzipped the unmolested copy, tried node install full, same traceback referring to bunzip2...

Renamed that dir and renamed the pre-this copy back to dtcyber-main, click node start shortcut, no prob...

I have sharex, I can record gif and MP4 from desktop or window.  I need to do it again anyway, I thought I screenshotted the traceback, but must not have.

**Comment by kej715** [comment]
If you have updated node.js, various node modules might also need to be
updated (e.g., maybe the unbzip2 library needs updating). This is
accomplished using "npm install", as described in the NOS2.8.7/README.md
file. Specifically, on Windows, you should execute "npm install" in the
automation, rje-station, stk, and webterm directories. On Unix/Linux, "make
-f Makefile.xxxx all" runs "npm install" where needed.

**Reply by KC5VDJ** [reply]
I'll give that a try tomorrow, thanks.  I did manage to duplicate and get that trace.

![GOM64_461NKqU71L.png](https://github.com/kej715/DtCyber/assets/136194499/d6340233-e12e-44c4-bb44-6caaee904ad3)

**Reply by KC5VDJ** [reply]
How embarrassing...   Thanks...  node install full running.

Thanks for the good work!  The penultimate 170 running the penultimate nos-2.

**Reply by wschaub** [reply]
The other thing you ran into when trying out the 875 I've also run into. when the memory size changes the UEM area isn't in the same place and even though we save a persistent copy of CM you will want to initialize ECS/UEM again by doing something like this after you've increased your memory size (by switching to an 875) 

../dtcyber manual 
select O on the console
select P on the console
D=YES
hit enter on the console
NEXT.
INITIALIZE,AL,5,6.
GO.
let the system come up and you should be able to run CYBIS and shut it down without the SECURED DEVICE warning coming up.  You should then be able to do a checkpoint system and shutdown and start the emulator normally with node start like before.

**Reply by KC5VDJ** [reply]
Okay, bring cybis up normal with CM at 4MW,
Produce the problem.
Init the CM.
How do I get rid of "AAAY ... WAIT. FOR PLATO CRASH DUMPS TO", in this instance hanging onto CP 7 in this instance, have yet to bring up cybis because this appears hung.  Idle doesn't work.  Is it safe to kill?

> The other thing you ran into when trying out the 875 I've also run into. when the memory size changes the UEM area isn't in the same place and even though we save a persistent copy of CM you will want to initialize ECS/UEM again by doing something like this after you've increased your memory size (by switching to an 875) 
> 
> ../dtcyber manual 
> select O on the console
> select P on the console
> D=YES
> hit enter on the console
> NEXT.
> INITIALIZE,AL,5,6.
> GO.
> let the system come up and you should be able to run CYBIS and shut it down without the SECURED DEVICE warning coming up.  You should then be able to do a checkpoint system and shutdown and start the emulator normally with node start like before.

**Comment by kej715** [comment]
In my simple test, I changed only the model from CY865 to CY875 and did not
change the memory size (it remained at 4000000), so it wasn't necessary to
re-initialize the two extended memories. The system deadstarted
successfully, and CYBIS started successfully.

**Reply by KC5VDJ** [reply]
Hmmm...  I'm having to install full again when iirc the user cybis was being moved to the 377777 user index and the move failed.  I have 'tail --retry -f LP5xx_C12_E5' of the build from the beginning, gnuwin util.  144K gzipped, 932K decompressed.
[install.out.txt.gz](https://github.com/kej715/DtCyber/files/12336856/install.out.txt.gz)

This was building with CM=20000000.

I honestly don't see how, getting this far, that this failed move could be a memory thing.  Ether-trash maybe, I've had to reget before..  currently doing full again, but with one MW again instead of four.

**Reply by KC5VDJ** [reply]
I did also forget to mention that the first deadstart after building in site.cfg changes (and all install deadstarts thereafter? The next one as well), that deadstart date hangs.  Looks like it rejects an invalid date requiring operator intervention for it and time.  The script picks up again after that.

I'll get back to you on that, as I am using Intel classic c 19.2, and an issue involving this is a known over-optimization thing, but I have nothing but Intel default at the moment, iirc.

> Hmmm...  I'm having to install full again when iirc the user cybis was being moved to the 377777 user index and the move failed.  I have 'tail --retry -f LP5xx_C12_E5' of the build from the beginning, gnuwin util.  144K gzipped, 932K decompressed.
> [install.out.txt.gz](https://github.com/kej715/DtCyber/files/12336856/install.out.txt.gz)
> 
> This was building with CM=20000000.
> 
> I honestly don't see how, getting this far, that this failed move could be a memory thing.  Ether-trash maybe, I've had to reget before..  currently doing full again, but with one MW again instead of four.
>

**Comment by kej715** [comment]
If entry of the date is failing at deadstart, then probably the 5 second
delay in the cyber.ini file is not sufficient for your system. The date and
time are entered via the [operator.nos287] section of the cyber.ini file.
The section looks like this:

[operator.nos287]
;----------------------------------------------------------------------
; This section is used by the default [cyber] configuration profile to
; automate entry of date and time on normal level 0 deadstarts and
; set up the operator command port.
;----------------------------------------------------------------------
set_operator_port 6662
enter_keys #2000#
enter_keys #5000#%year%%mon%%day%
enter_keys #3000#%hour%%min%%sec%


the integers surrounded by '#' characters specify delays in milliseconds.
You can try increasing 5000 to a greater value.

While it -should- be possible to execute "node install full" successfully,
it's certainly not necessary to do a full install in order to produce a
full NOS 2.8.7 system. The system image that is installed when you execute
"node install" (without "full") was produced by executing "node install
full", so there's no compelling reason to perform a full install yourself
unless you want to see how the "sausage is made", and you're willing to
wait for that long process to complete.

If you want to customize your system, you can specify customizations in the
site.cfg file, and "node install" (without "full") will apply them
automatically after installing the generic full system.

**Reply by KC5VDJ** [reply]
That might have done it..  changed from 5 to 7.5 secs and from 3 to 5 secs.  Zero intervention needed on post-site.cfg deadstart.  We'll see, but I'll say those timeouts work for now.

**Reply by KC5VDJ** [reply]
Guess not.  The post cybis unpack deadstart still required intervention.

**Reply by KC5VDJ** [reply]
Ok, and this time it lost its place and froze after the dtcyber console message "move cybis from ui 1 to ui 377777".  Three bombs at this point in one fashion or another.  Exact point.

My only other alteration of any kind would be that I have commented out the webterm server in the .ini.

**Comment by kej715** [comment]
Regarding where to drop a large file ... if you have a DropBox or Google
you can use DropBox, Google Drive, or similar account, you can upload the
file there and provide a link.

**Comment by kej715** [comment]
I'm sorry that you're having trouble with this. Due to the design of the
Cyber console and DtCyber's current implementation of it, the
implementation of the console automation must rely on timing in various
areas. For example, it relies on specific delays during deadstart (as
you've seen in date/time entry), it relies on a specific delay when it
launches DIS via "X.DIS." before issuing DIS commands, etc. These delays
were chosen based upon experience running DtCyber and the automation on
various host systems (e.g., Mac and Windows laptops). Apparently, your host
system is significantly slower than the reference systems, or it is more
burdened by background activities, so the delays currently programmed into
the automation aren't sufficient for your machine.

You've seen some success tweaking the initial delay at deadstart affecting
date/time entry. You can also try tweaking delays in the
automation/DtCyber.js module. For example, the dis() method in DtCyber.js
module specifies a 2000 millisecond delay after launching DIS via "X.DIS."
You might try increasing that delay. The "move cybis from ui 1o 377777"
step uses DIS (as do many other installation steps).

The ready-to-run packages installable via "node install" (without "full"
option) for the various OS'es were created on the reference systems using
"node install full". They were created for a couple of reasons:

   1. to relieve hobbyists from being required to perform long, tedious
   installs via "node install full"
   2. to prevent hobbyists from possibly encountering timing issues when
   attempting to run "node install full" on host machines, or in host
   environments, that might not provide the level of performance provided by
   the reference machines

**Comment by philclaridge** [comment]
Here are the settings I use to enter the time onto a slow machine (NOS2.8.7 on Raspberry Pi 2 Zero)

                set_operator_port 6662 (unchanged)
                set_key_wait_interval 500
                enter_keys #4000#
                enter_keys #25000#%year%%mon%%day%
                enter_keys #3000#%hour%%min%%sec%

**Comment by KC5VDJ** [comment]
Thanks, I'll give it a try again today sometime when life gets out of the
way...  tried again last night, using the mickeysoft backend and the virgin
solution, and it still bombed, interestingly, for a fifth time at pretty
much the same spot, moving the Plato user index, again.  This rules out the
compiler for that but...seeing that the compiler wasn't doing it, and since it appears no new files have been added to the vs solution so far as I can tell, I'm actually going to try my Intel compiler settings which were working.  If it works, I'll upload the converted solution files.  They do require Intel parallel studio, and it looks as if Intel doesn't charge for it anymore...

I'll give your numbers a try.  How's the performance on a pi zero?  I have several, and have been thinking of this for that.

**Reply by KC5VDJ** [reply]
No real changes other than an optimized Intel recompile, now it is past the whole cybis users thing, doing opt/cybis.post now.  And the speed....

Let's see...

Del disks\DQ*
Del persistence\*
Copy over solution files from running 865 (Intel 19.2, best optimization I've found...
Node install full

Yup, no changes other than recompile....

Not declaring success yet, but it's a good omen.

**Reply by KC5VDJ** [reply]
Okay, never say it's all good until the fat lady sings....check.![8P4iIVzY7D.png](https://github.com/kej715/DtCyber/assets/136194499/907b29c5-e69f-487f-9a1c-5a5517e7b00b)

Ugh.

**Comment by kej715** [comment]
If. you set the CM at the full 20000000 (octal) words for an 875, than
you'll need to define how large UEM is, e.g., by defining
"EQ005=DE,ET=EM,SZ=10000." in the equipment deck at deadstart, and then
initializing UEM and ESM using "INITIALIZE,AL,5,6." Setting "SZ=1000000" in
the definition of the UEM equipment gives UEM half of the memory (i.e., 2M
words, or 16M bytes), which leaves the other half of memory (i.e., 2M
words, or 16M bytes) for CM.

To get rid of job AAAY which is asking for assignment of tape for a CYBIS
dump, simply enter the DSD command "DROP,AAAY.".

**Reply by KC5VDJ** [reply]
I'll try that tomorrow at some point.  Thanks.

**Comment by kej715** [comment]
If the DROP command produced "INCORRECT ENTRY", then either the JSN
provided on the DROP command did not match the JSN of any running job in
the system, or the JSN provided was the JSN of a subsystem. Please provide
a screenshot showing the job that is waiting for the crash dump. We can
then let you know exactly what DSD command to issue in order to drop the
job or assign it a tape or null equipment.
